pypeit/wavecalib.py

```diff
@@ -6,44 +6,230 @@
+            msgs.error("Bad parsing of the MasterFlat")
+        # Finish
+        _d['wv_fits'] = np.asarray(list_of_wave_fits)
+        return _d, dm_version_passed, dm_type_passed, parsed_hdus
+
+    @property
+    def par(self):
+        return json.loads(self.strpar)
+
+    def chk_synced(self, slits):
+        """
+        Confirm the slits in WaveCalib are aligned to that in SlitTraceSet
+
+        Barfs if not
+
+        Args:
+            slits (:class:`pypeit.slittrace.SlitTraceSet`):
+
+        """
+        if not np.array_equal(self.spat_id, slits.spat_id):
+            msgs.error("Your wvcalib solutions are out of sync with your slits.  Remove Masters and start from scratch")
+
+    def build_waveimg(self, tilts, slits, spat_flexure=None):
+        """
+        Main algorithm to build the wavelength image
+
+        Only applied to good slits, which means any non-flagged or flagged
+         in the exclude_for_reducing list
+
+        Args:
+            tilts (`numpy.ndarray`_):
+                Image holding tilts
+            slits (:class:`pypeit.slittrace.SlitTraceSet`):
+            spat_flexure (float, optional):
+
+        Returns:
+            `numpy.ndarray`_: The wavelength image.
+        """
+        # Setup
+        #ok_slits = slits.mask == 0
+        bpm = slits.mask.astype(bool)
+        bpm &= np.logical_not(slits.bitmask.flagged(slits.mask, flag=slits.bitmask.exclude_for_reducing))
+        ok_slits = np.logical_not(bpm)
+        #
+        image = np.zeros_like(tilts)
+        slitmask = slits.slit_img(flexure=spat_flexure, exclude_flag=slits.bitmask.exclude_for_reducing)
+
+        slit_spat_pos = slits.spatial_coordinates(flexure=spat_flexure)
+
+        # If this is echelle print out a status message and do some error checking
+        if self.par['echelle']:
+            msgs.info('Evaluating 2-d wavelength solution for echelle....')
+            # TODO UPDATE THIS!!
+            #if len(wv_calib['fit2d']['orders']) != np.sum(ok_slits):
+            #    msgs.error('wv_calib and ok_slits do not line up. Something is very wrong!')
+
+        # Unpack some 2-d fit parameters if this is echelle
+        for islit in np.where(ok_slits)[0]:
+            slit_spat = slits.spat_id[islit]
+            thismask = (slitmask == slit_spat)
+            if not np.any(thismask):
+                msgs.error("Something failed in wavelengths or masking..")
+            if self.par['echelle']:
+                # # TODO: Put this in `SlitTraceSet`?
+                # evaluate solution --
+                image[thismask] = self.wv_fit2d.eval(
+                    tilts[thismask], x2=np.full_like(tilts[thismask], slits.ech_order[islit]))
+                image[thismask] /= slits.ech_order[islit]
+            else:
+                iwv_fits = self.wv_fits[islit]
+                image[thismask] = iwv_fits.pypeitfit.eval(tilts[thismask])
+        # Return
+        return image
+
+
+
+
+class BuildWaveCalib(object):
```

Comment:
You don't have to subclass from `object` in Python 3.  I've been steadily removing these.

Reply:
gone